lib/msal-common/src/account/AccountInfo.ts

```diff
@@ -35,9 +36,100 @@ export type AccountInfo = {
     };
     nativeAccountId?: string;
     authorityType?: string;
+    tenantProfiles?: Map<string, TenantProfile>;
+};
+
+/**
+ * Account details that vary across tenants for the same user
+ *
+ * - tenantId               - Full tenant or organizational id that this account belongs to
+ * - localAccountId         - Local, tenant-specific account identifer for this account object, usually used in legacy cases
+ * - name                   - Full name for the account, including given name and family name
+ * - isHomeTenant           - True if this is the home tenant profile of the account, false if it's a guest tenant profile
+ */
+export type TenantProfile = Pick<
+    AccountInfo,
+    "tenantId" | "localAccountId" | "name"
+> & {
+    isHomeTenant?: boolean;
 };
 
 export type ActiveAccountFilters = {
     homeAccountId: string;
     localAccountId: string;
+    tenantId?: string;
 };
+
+/**
+ * Returns true if tenantId matches the utid portion of homeAccountId
+ * @param tenantId
+ * @param homeAccountId
+ * @returns
+ */
+export function tenantIdMatchesHomeTenant(
+    tenantId?: string,
+    homeAccountId?: string
+): boolean {
+    return (
+        !!tenantId &&
+        !!homeAccountId &&
+        tenantId === homeAccountId.split(".")[1]
+    );
+}
+
+export function buildTenantProfileFromIdTokenClaims(
+    homeAccountId: string,
+    idTokenClaims: TokenClaims
+): TenantProfile {
+    const { oid, sub, tid, name, tfp, acr } = idTokenClaims;
+
+    const tenantId = tid || tfp || acr || "";
```

Comment:
Please add a comment explaining the meaning of these claims / reasoning behind this precedence